server/container_create_linux.go

```diff
@@ -408,6 +409,9 @@ func (s *Server) createSandboxContainer(ctx context.Context, ctr ctrfactory.Cont
 		if resources != nil {
 			specgen.SetLinuxResourcesCPUPeriod(uint64(resources.CpuPeriod))
 			specgen.SetLinuxResourcesCPUQuota(resources.CpuQuota)
+			if runtimehandlerhooks.ShouldCPUQuotaBeDisabled(sb.Annotations()) {
```

Comment:
Notice the original place actually checked that it is disabling quota for a guaranteed QoS class container with dedicated cpus. This code does not. So it makes a pretty big security hole by disabling resource accounting for burstable and best effort containers as well.

Reply:
good catch. I updated the condition to check for those